modules/saliency/include/opencv2/saliency/saliencySpecializedClasses.hpp

```diff
@@ -405,6 +405,173 @@ class CV_EXPORTS ObjectnessBING : public Objectness
 
 //! @}
 
+
+
+/**
+ * \brief Objectness algorithms based on [4]
+ * [4] Zitnick, C. Lawrence, and Piotr Dollár. "Edge boxes: Locating object proposals from edges." Computer Vision–ECCV 2014
+ */
+
+/** @brief the EdgeBoxes algorithm from @cite EdgeBoxes
+ */
+class CV_EXPORTS ObjectnessEdgeBoxes : public Objectness
+{
+public:
+
+	const int EDGE_IMAGE = 0;
+	const int ORIGINAL_IMAGE = 1;
+
+  ObjectnessEdgeBoxes();
+  virtual ~ObjectnessEdgeBoxes();
+
+  /** @brief Return the list of the rectangles' objectness value,
+
+    in the same order as the *vector\<Vec4i\> objectnessBoundingBox* returned by the algorithm (in
+    computeSaliencyImpl function). The bigger value these scores are, it is more likely to be an
+    object window.
+     */
+  std::vector<float> getobjectnessValues();
+
+  /** @brief This is a utility function that allows to set an arbitrary path in which the algorithm will save the
+    optional results
+
+    (ie writing on file the total number and the list of rectangles returned by objectess, one for
+    each row).
+    @param resultsDir results' folder path
+     */
+  void setBBResDir( std::string resultsDir );
+
+  bool computeSaliencyMap( Mat &edgeImage, Mat &orientationImage, Mat &resultImage);
+
+  virtual bool ObjectnessEdgeBoxes::computeSaliency(InputArray image, OutputArray saliencyMap);
+  virtual bool ObjectnessEdgeBoxes::computeSaliency(InputArray image, OutputArray saliencyMap, const int mode);
+
+protected:
+  /** @brief Performs all the operations and calls all internal functions necessary for the
+  accomplishment of the EdgeBoxes algorithm.
+
+    @param image input image. According to the needs of this specialized algorithm, the param image is a
+    single *Mat*
+    @param objectnessBoundingBox objectness Bounding Box vector. According to the result given by this
+    specialized algorithm, the objectnessBoundingBox is a *vector\<Vec4i\>*. Each bounding box is
+    represented by a *Vec4i* for (minX, minY, maxX, maxY).
+     */
+  bool computeSaliencyImpl( InputArray image, OutputArray objectnessBoundingBox );
+
+private:
+
+
+	bool _bScoresReady = false;
```

Comment:
This variable should be initialized in class constructor to conform to C++03 standard.